src/Datadog.Trace/Tracer.cs

```diff
@@ -440,19 +440,26 @@ private static string GetApplicationName()
 
         private static IStatsd CreateDogStatsdClient(TracerSettings settings, string serviceName, int port)
         {
-            var frameworkDescription = FrameworkDescription.Create();
-
-            string[] constantTags =
+            try
             {
-                "lang:.NET",
-                $"lang_interpreter:{frameworkDescription.Name}",
-                $"lang_version:{frameworkDescription.ProductVersion}",
-                $"tracer_version:{TracerConstants.AssemblyVersion}",
-                $"service_name:{serviceName}"
-            };
+                var frameworkDescription = FrameworkDescription.Create();
+                string[] constantTags =
+                {
+                    "lang:.NET", $"lang_interpreter:{frameworkDescription.Name}",
+                    $"lang_version:{frameworkDescription.ProductVersion}",
+                    $"tracer_version:{TracerConstants.AssemblyVersion}",
+                    $"service_name:{serviceName}"
+                };
+
+                var statsdUdp = new StatsdUDP(settings.AgentUri.DnsSafeHost, port, StatsdConfig.DefaultStatsdMaxUDPPacketSize);
+                return new Statsd(statsdUdp, new RandomGenerator(), new StopWatchFactory(), prefix: string.Empty, constantTags);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, $"Unable to instantiate {nameof(Statsd)} client.");
+            }
 
-            var statsdUdp = new StatsdUDP(settings.AgentUri.DnsSafeHost, port, StatsdConfig.DefaultStatsdMaxUDPPacketSize);
-            return new Statsd(statsdUdp, new RandomGenerator(), new StopWatchFactory(), prefix: string.Empty, constantTags);
+            return new NoOpStatsd();
```

Comment:
Just a nit, but if this line is moved inside the `catch`, it's clearer that it happens only if an exception is raised.